gtest_parallel.py

```diff
@@ -170,12 +170,16 @@ def __init__(self, test_binary, test_name, test_command, execution_number,
     self.log_file = Task._logname(self.output_dir, self.test_binary,
                                   test_name, self.execution_number)
 
-  def __lt__(self, other):
-    if self.last_execution_time is None:
-      return True
-    if other.last_execution_time is None:
-      return False
-    return self.last_execution_time > other.last_execution_time
+  def __sorting_key(self):
+    # Unseen or failing tests (both missing execution time) take precedence over
+    # execution time.
+    return (1 if self.last_execution_time is None else 0,
+            self.last_execution_time)
```

Comment:
It may be best to put the minus sign here instead of below. Otherwise, you have to explain here why 1 gets sorted before 0.

Reply:
since sorted(foo) is normally in increasing order, should we instead change the sorted call to `sorted(foo, reverse=True)`?

Reply:
That's clearer in my opinion.

Reply:
Done, I'll wait until your tests are in to submit this regardless.